hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -192,7 +193,7 @@ object HoodieSparkSqlWriter {
             }
 
             // Get list of partitions to delete
-            val partitionsToDelete = genericRecords.map(gr => keyGenerator.getKey(gr).getPartitionPath).toJavaRDD().distinct().collect()
+            val partitionsToDelete = genericRecords.map(gr => gr.get(partitionColumns).toString).toJavaRDD().distinct().collect()
```

Comment:
not sure if we can generalize this. partitionColumns is a comma separate list of partition col field names if Complex or CustomeKeyGen is used.

Reply:
guess we have to add another config here. 
"hoodie.datasource.write.partitions.to.delete" // goes into DataSourceWriteOptions

when user sets this config with "delete_partitions" operation, we can parse this config value. this should refer to comma separated list of partitions to delete. 
If this config is not set, we will go the usual route of fetching partition cols from keyGen and then using keyGen to determine the partitions to be deleted from partition path for records. Basically apache/master as is. 

Do remember in the first case, we should not be using 
```
HoodieWriterUtils.getPartitionColumns(keyGenerator)
```
to fetch partition col fields.

Let me know what do you think?

Reply:
First of all, thank you very much for your attention to this issue in your busy work.
I will try more unit tests to verify and improve this problem.
I think your idea is good. I will conduct more verifications. If I find other problems and questions, I will feedback them. When I am finished, I will request your review here.

Reply:
sure.